tools/pv_tree_generator/build_tree.py

```diff
@@ -0,0 +1,174 @@
+          #build the branches recursively
+          for child in child_pos:
+            branch = build_tree_recursive(child, level + 1, pop_obs_spec,
+                                        stat_vars, show_all, value_ui_node)
+            if branch['children']:
+              value_blob['children'].append(branch)
+            value_blob['sv_set'] |= branch['sv_set']
+            del branch['sv_set']
+        value_blob['count'] = len(value_blob['sv_set'])
+      
+    result['children'] = text_format.filter_and_sort(pdiff, result['children'], show_all)
+    #update the count
+    if result['children']:
+      for child in result['children']:
+        result['sv_set'] |= child['sv_set']
+        del child['sv_set']
+    result['count'] = len(result['sv_set'])
+    return result
+    
+def build_tree(v, pop_obs_spec, stat_vars, show_all):
+    """Build the tree for each vertical."""
+    
+    #vertical as the root
+    root = {
+        'show': 'yes',
+        'selected': 'no',
+        'expanded': 'no',
+        'argString': 'top',
+        'title': text_format.format_title(v),
+        'type': 'Property',
+        'count': 0, #count of child nodes
+        'children': [],
+        'sv_set': set(),#used for counting child nodes
+    }
+    
+    #specs with 0 constaints are of type "value", as the level 1 children of root
+    for pos in pop_obs_spec[0]:
+      ui_node = util.UiNode(pos, {}, False)
+      root['children'].append({
+        'populationType': ui_node.pop_type,
+        'show': 'yes',
+        'selected': 'no',
+        'expanded': 'no',
+        'argString': ui_node.arg_string,
+        'title': text_format.format_title(ui_node.text),
+        'type': 'value',
+        'children': [],
+        'count': 1,
+      })
+      root['count'] += 1
+    
+    #build specs with >= 1 constraints recursively
+    for pos in pop_obs_spec[1]:
+      child = build_tree_recursive(pos, 1, pop_obs_spec, stat_vars, show_all)
+      # For certain branch, we would like to put them under 0 pv nodes:
+      if (pos.pop_type in ['EarthquakeEvent', 'CycloneEvent', 'MortalityEvent']):
+        for pv0 in root['children']:  # hoist logic will break if multiple 0 pv
+          if pv0['argString'] == '{},count'.format(pos.pop_type):
+            pv0['children'].append(child)
+            if 'sv_set' not in pv0:
+              pv0['sv_set'] = set()
+            pv0['sv_set'] |= child['sv_set']
+            break
+      else:
+        root['children'].append(child)
+      root['sv_set'] |= child['sv_set']
+      del child['sv_set']
+     
+     
+    #update the count
+    for pv0 in root['children']:
+      if 'sv_set' in pv0:
+        pv0['count'] += len(pv0['sv_set'])
+        del pv0['sv_set']
+    root['count'] += len(root['sv_set'])
+    del root['sv_set']
+    
+    return root
+     
+def main():
+    pop_obs_spec = util._read_pop_obs_spec()
+    stat_vars = util._read_stat_var()
+    f_hierachy = open("./hierarchy.js","w")
```

Comment:
we don't need the .js file anymore, can remove code about it

Reply:
done